streams/src/main/java/org/apache/kafka/streams/kstream/internals/BaseJoinProcessorNode.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.kstream.ValueJoiner;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+
+/**
+ * Utility base class containing the common fields between
+ * a Stream-Stream join and a KTable-KTable join
+ */
+abstract class BaseJoinProcessorNode<K, V, V1, V2, VR> extends StreamGraphNode<K, V> {
+
+    private final ProcessorSupplier<K, V1> joinThisProcessSupplier;
+    private final ProcessorSupplier<K, V2> joinOtherProcessSupplier;
+    private final ProcessorSupplier<K, VR> joinMergeProcessor;
+    private final ValueJoiner<? super V, ? super V1, ? extends V2> valueJoiner;
+    private final String joinThisProcessorName;
+    private final String joinOtherProcessorName;
+    private final String joinMergeProcessorName;
+    private final String thisJoinSide;
+    private final String otherJoinSide;
+
+
+    BaseJoinProcessorNode(final String predecessorNodeName,
+                          final String name,
+                          final ValueJoiner<? super V, ? super V1, ? extends V2> valueJoiner,
```

Comment:
This is not a change suggestion comment: when we are constructing the node, I think besides the first three parameters all other fields are not known yet, and they will only be determined later. So I'm curious how we would pass these parameters in the following PRs. We can discuss about this more in that PR.

Reply:
Should we also include the `Joined` in this graph node as well?

Reply:
ack

Reply:
Ack.